iceoryx_posh/test/moduletests/test_popo_waitset.cpp

```diff
@@ -33,317 +34,422 @@ using namespace iox::units::duration_literals;
+            {
+                m_triggerHandle.erase(m_triggerHandle.begin() + i);
+            }
         }
+    }
+
+    static void triggerCallback1(WaitSet_test* const waitset)
+    {
+        waitset->m_triggerCallbackArgument1 = waitset;
+    }
 
-        /// @note members reside in ChunkQueueData in SHM
-        bool m_wasTriggered{false};
-        ConditionVariableData* m_condVarPtr{nullptr};
-    };
+    static void triggerCallback2(WaitSet_test* const waitset)
+    {
+        waitset->m_triggerCallbackArgument2 = waitset;
+    }
 
-    ConditionVariableData m_condVarData;
-    WaitSetMock m_sut{&m_condVarData};
-    vector<MockSubscriber, iox::MAX_NUMBER_OF_CONDITIONS_PER_WAITSET> m_subscriberVector;
+    void SetUp(){};
 
-    iox::posix::Semaphore m_syncSemaphore =
-        iox::posix::Semaphore::create(iox::posix::CreateUnnamedSingleProcessSemaphore, 0u).value();
+    void TearDown(){};
+};
 
-    void SetUp()
-    {
-        while (m_subscriberVector.size() != m_subscriberVector.capacity())
-        {
-            m_subscriberVector.emplace_back();
-        }
-    };
+TEST_F(WaitSet_test, AcquireTriggerOnceIsSuccessful)
+{
+    EXPECT_FALSE(acquireTrigger(m_sut, 0)->has_error());
+}
+
+TEST_F(WaitSet_test, AcquireMultipleTriggerIsSuccessful)
+{
+    auto trigger1 = acquireTrigger(m_sut, 10);
+    auto trigger2 = acquireTrigger(m_sut, 11);
+    auto trigger3 = acquireTrigger(m_sut, 12);
 
-    void TearDown()
+    EXPECT_FALSE(trigger1->has_error());
+    EXPECT_FALSE(trigger2->has_error());
+    EXPECT_FALSE(trigger3->has_error());
+}
+
+TEST_F(WaitSet_test, AcquireMaximumAllowedTriggersIsSuccessful)
+{
+    iox::cxx::vector<expected<TriggerHandle, WaitSetError>*, iox::MAX_NUMBER_OF_TRIGGERS_PER_WAITSET> trigger;
+    for (uint64_t i = 0; i < iox::MAX_NUMBER_OF_TRIGGERS_PER_WAITSET; ++i)
     {
-        m_sut.detachAllConditions();
-        m_subscriberVector.clear();
-        ConditionVariableWaiter waiter{&m_condVarData};
-        waiter.reset();
-    };
-};
+        trigger.emplace_back(acquireTrigger(m_sut, 1 + i));
+        EXPECT_FALSE(trigger.back()->has_error());
+    }
+}
 
-TEST_F(WaitSet_test, AttachSingleConditionSuccessful)
+TEST_F(WaitSet_test, AcquireMaximumAllowedPlusOneTriggerFails)
 {
-    EXPECT_FALSE(m_sut.attachCondition(m_subscriberVector.front()).has_error());
-    EXPECT_TRUE(m_sut.isConditionAttached(m_subscriberVector.front()));
+    iox::cxx::vector<expected<TriggerHandle, WaitSetError>*, iox::MAX_NUMBER_OF_TRIGGERS_PER_WAITSET> trigger;
+    for (uint64_t i = 0; i < iox::MAX_NUMBER_OF_TRIGGERS_PER_WAITSET; ++i)
+    {
+        trigger.emplace_back(acquireTrigger(m_sut, 5 + i));
+    }
+    auto result = acquireTrigger(m_sut, 0);
+    ASSERT_TRUE(result->has_error());
+    EXPECT_THAT(result->get_error(), Eq(WaitSetError::TRIGGER_VECTOR_OVERFLOW));
 }
 
-TEST_F(WaitSet_test, AttachConditionToDifferentWaitsetDetachesConditionFromOrigin)
+TEST_F(WaitSet_test, AcquireSameTriggerTwiceResultsInError)
```

Comment:
This would be a problem if we want to delete the trigger with internal id 0? There would be two of them, in principle both could be deleted. I do not really see why acquiring duplicate triggers has to be forbidden but I guess it is also not too useful. 

In my book you would be notified for both on wake up and so on.

Those restrictions need to be made clear in the readme if they are not already.

Reply:
Checked it. It is already stated in the doxygen documentation of the `acquireTrigger` method